### PR TITLE
First PR (of a total of 3) to address #491

### PR DESCRIPTION
**Pull request by maugustosilva**
* The `agent` will now retrieve TPM 2.0 preboot event log and will ship
it to the `verifier`. The latter will do nothing with it (just receives
and ignores it)
* Adds a new `keylime_tenant` CLI option, `--mb_refstate` (that was a
request from @mpeters) to signal the need for the agent to collect the
the event log. At this point, `keylime_tenant` do not actually process
this option (it shall be processed in a manner similar to IMA in a
future PR), it is simply used to signal the agent that "tpm2 preboot
event log" should be read.
* Hard-coded PCRs [0-9], 14 and path
/sys/kernel/security/tpm0/binary_bios_measurements for the event log
* The `verifier` will output a debug logging message stating that it
received the "tpm2 event log", and will send it down to the `check_pcrs`
method, but there is no actual processing there (it will be done in
subsequent PR)

Signed-off-by: Marcio Silva <marcios@us.ibm.com>

### CODE REVIEW

**Comment by maugustosilva** [comment]
Before @lukehinds even asks :-), here is the link to the enhancement: https://github.com/keylime/enhancements/issues/40

**Comment by maugustosilva** [comment]
Complete doc for enhancement submitted as a [PR](https://github.com/keylime/enhancements/pull/41) on the appropriate repo.

File: keylime/cloud_verifier_common.py
Comment (mpeters) [inline keylime/cloud_verifier_common.py:217]: Just to not confuse this PR, can we remove this commented out code?

File: keylime/tenant.py
Comment (mpeters) [inline keylime/tenant.py:204]: We don't need to prefix method/function variables with underscore. That convention in python is just for package level things so they aren't exported.

File: keylime/tenant.py
Comment (mpeters) [inline keylime/tenant.py:254]: We're doing this check twice (line 209 above). I know it's in different methods, but just wanted to check that we need to have it in both places. And if we do, maybe it should be in it's own helper method?

File: keylime/tenant.py
Comment (mpeters) [inline keylime/tenant.py:270]: Let's remove this commented code in this first PR

**Review by mpeters — CHANGES_REQUESTED** [review]
Some relatively small changes requested

**Review by mpeters — APPROVED** [review]
This LGTM. I'll hold off on merging until @lukehinds has a chance to review early next week

File: keylime/cloud_verifier_common.py
Comment (lukehinds) [inline keylime/cloud_verifier_common.py:217]: Just curious what the empty dict is for?

File: keylime/config.py
Comment (lukehinds) [inline keylime/config.py:299]: not a blocker by any means, but I wonder if this might be useful exposed in keylime.conf. I guess not and you would know better here.

File: keylime/tenant.py
Comment (lukehinds) [inline keylime/tenant.py:401]: Just to clarify my understanding, does mean users can cherry pick PCRs to either tpm_policy or measure boot (mb)?

File: keylime/tenant.py
Comment (lukehinds) [inline keylime/tenant.py:402]: I would change this to `sys.exit(1)` as it defaults to `0` which is success.  This would be useful then for CI systems etc that might wrap around the tenant CLI.

https://docs.python.org/3/library/sys.html#sys.exit

File: keylime/tpm/tpm1.py
Comment (lukehinds) [inline keylime/tpm/tpm1.py:742]: be good to understand the empty dicts again

File: keylime/tpm/tpm_abstract.py
Comment (lukehinds) [inline keylime/tpm/tpm_abstract.py:241]: This could perhaps steer the user more towards the reason why. If I saw this message I would think 'ok, when will it be processed and why?'.

**Review by lukehinds — CHANGES_REQUESTED** [review]
I think the patch is in good order, just some comments and one change request.

File: keylime/cloud_verifier_common.py
Comment (maugustosilva) [reply keylime/cloud_verifier_common.py:217]: in the subsequent PRs, it will be changed to the "measured boot reference state" supplied by the `tenant` CLI

File: keylime/config.py
Comment (maugustosilva) [reply keylime/config.py:299]: I thought about it, and decided to keep all values "hard-coded" mostly because - very much like IMA - "measured boot" refers to a very specific set of PCRs. Granted, there is some latitude on PCR 14 (which is updated only when secure booting), but I plan to revisit it after the basic set (of 3) PRs gets merged.

File: keylime/tenant.py
Comment (maugustosilva) [reply keylime/tenant.py:401]: Correct

File: keylime/tenant.py
Comment (maugustosilva) [reply keylime/tenant.py:402]: good catch! I have done this now, and later will I make a PR to have non-zero exit code for other errors (unrelated to this PR)

File: keylime/tpm/tpm1.py
Comment (maugustosilva) [reply keylime/tpm/tpm1.py:742]: in the subsequent PRs, it will be changed to the "measured boot reference state" supplied by the tenant CLI

File: keylime/tpm/tpm_abstract.py
Comment (galmasi) [reply keylime/tpm/tpm_abstract.py:241]: the message will disappear when PR2 shows up. Probably should have marked this as "temporary"

**Review by lukehinds — APPROVED** [review]

**Comment by lukehinds** [comment]
you should have merge rights @maugustosilva , if not ping me or @mpeters

**Pull request merged**
